server/routes/overview/overviewController.test.ts

```diff
@@ -41,12 +42,16 @@ describe('overviewController', () => {
   const timelineService = {
     getTimeline: jest.fn(),
   }
+  const prisonService = {
+    getPrisonByPrisonId: jest.fn(),
```

Comment:
It would be good to see the relevant tests in this test class make some assertions on this mock and/or stub its response.
What I mean by that is that injecting the mock into the controller is fine (and indeed you needed to do that otherwise it doesnt compile properly). But _why_ does the controller class need the Prison Service? What is is doing with it? Is it doing the right / expected thing with it?

You are making use of the Prison Service in the `getSupportNeedsView` method of the controller, so what you want to do is find the relevant tests in this test class, then setup this mock. IE. when the controller method runs it will invoke `prisonService.getPrisonByPrisonId` at least twice (at least once for the health and support needs, and at least once for the neurodiversity stuff). You need to set the mock up so that it returns a `Prison`
You then also need to assert (in the `// Then` section) that the mock was called the expected number of times and with the expected arguments (ie. in this case the prison ID and token)